pennylane/transforms/qcut.py

```diff
@@ -210,3 +213,121 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
             order += 1
 
     return graph
+
+
+# pylint: disable=too-many-branches
+def contract_tensors(
+    tensors: Sequence,
+    communication_graph: MultiDiGraph,
+    prepare_nodes: Sequence[Sequence[PrepareNode]],
+    measure_nodes: Sequence[Sequence[MeasureNode]],
+    use_opt_einsum: bool = False,
+):
+    """Contract tensors according to the edges specified in the communication graph.
+
+    This operation is differentiable. The ``prepare_nodes`` and ``measure_nodes`` arguments are both
+    sequences of size ``len(communication_graph.nodes)`` that describe the order of indices in the
+    ``tensors`` with respect to to the :class:`~.PrepareNode` and :class`~.MeasureNode` edges in the
+    communication graph.
+
+    Args:
+        tensors (Sequence): the tensors to be contracted
+        communication_graph (MultiDiGraph): the communication graph determining connectivity between
+            the tensors
+        prepare_nodes (Sequence[PrepareNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of preparation indices in each tensor
+        measure_nodes (Sequence[MeasureNode]): a sequence of size ``len(communication_graph.nodes)``
+            that determines the order of measurement indices in each tensor
+        use_opt_einsum (bool): Determines whether to use the
+            [opt_einsum](https://dgasmith.github.io/opt_einsum/) package. This package is useful for
+            tensor contractions of large networks but must be installed separately using, e.g.,
+            ``pip install opt_einsum``.
+
+    Returns:
+        float or array-like: the result of contracting the tensor network
+
+    **Example**
+
+    .. note::
+
+        This function is designed for use as part of the circuit cutting workflow. Check out the
+        :doc:`transforms </code/qml_transforms>` page for more details.
+
+    We first set up the tensors and their corresponding :class:`~.PrepareNode` and
+    :class`~.MeasureNode` orderings:
+
+    .. code-block:: python
+
+        t = [np.arange(4), np.arange(4, 8)]
+        p = [[], [qcut.PrepareNode(wires=0)]]
+        m = [[qcut.MeasureNode(wires=0)], []]
```

Comment:
slightly more descriptive variable names could be helpful here!

Reply:
Fixed!